Sources/TokamakCore/Fiber/AlignmentID.swift

```diff
@@ -17,7 +17,27 @@
 
 import Foundation
 
+/// Used identify an alignment guide.
+///
+/// Typically, you would define an alignment guide inside
+/// an extension on `HorizontalAlignment` or `VerticalAlignment`:
+///
+///     extension HorizontalAlignment {
+///       private enum MyAlignmentGuide: AlignmentID {
+///         static func defaultValue(in context: ViewDimensions) -> CGFloat {
+///           return 10.0
+///         }
+///       }
+///       public static let myAlignmentGuide = Self(MyAlignmentGuide.self)
+///     }
+///
+/// Which you can then use with the `alignmentGuide` modifier:
+///
+///     Text("Hello, world!")
+///       .alignmentGuide(.myAlignmentGuide) { $0[.trailing] - 3 }
```

Comment:
Hm, to be honest I've never used `AlignmentID` before. Just tried this code in real Swift UI and it doesn't have any impact on `Text` positioning. Is there some context in which `.alignmentGuide` works? I'm just having this plain `Text` with `.alignmentGuide` modifier with a custom AG. Tried both in a stack and as a root view, same thing, no effect.

Reply:
I have used them before, so this example is probably just bad. I didn't test it out, I'll come up with a working sample.

Reply:
I updated the sample to show one scenario they could be used. You can use the custom guide as the alignment for a `VStack`. Then the `alignmentGuide` modifier will have an effect if you specify the same `AlignmentID` as the one to modify, because that's the one the VStack layout is reading.

We don't have the `alignmentGuide` modifier yet, but it should work the same once we implement it, and `AlignmentID` is used internally for all the builtin alignments.